extension/intents/saving/saving.js

```diff
@@ -18,7 +20,12 @@ this.intents.saving = (function() {
       const { html, metadata } = await browser.tabs.sendMessage(activeTab.id, {
         type: "freezeHtml",
       });
-      const filename = makeFilename(metadata.title, metadata.url, ".html");
+      const name = context.slots.name;
```

Comment:
The name should be cleaned with these same regexes that the page title is cleaned with: https://github.com/mozilla/firefox-voice/blob/master/extension/intents/saving/saving.js#L58-L59 (best to extract those into a function).

Unfortunately the download API doesn't automatically clean filenames.

Reply:
Ok. Thanks for the feedback.